server/documents/document.js

```diff
@@ -44,6 +48,291 @@ Meteor.methods({
       });
     }
   },
+  'Document.fork'(args) {
+    check(args, {
+      documentId: String,
+    });
+
+    const user = Meteor.user(User.REFERENCE_FIELDS());
+
+    // TODO: Check fork permissions.
+    if (!user || !user.hasPermission(Document.PERMISSIONS.CREATE)) {
+      throw new Meteor.Error('unauthorized', "Unauthorized.");
+    }
+    const doc = Document.documents.findOne({
+      _id: args.documentId,
+    });
+
+    const {contentKey} = doc;
+
+    const createdAt = new Date();
+    const forkContentKey = Content.Meta.collection._makeNewID();
+
+    // Add forkContentKey to the existing contents.
+    Content.documents.update(
+      {
+        contentKeys:
+        {
+          $elemMatch: {
```

Comment:
I do not think you need `$elemMatch` here. You are matching only against one field.